source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -411,6 +411,11 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
       if (cluster_) {
         config_->incCounter(cluster_->statsScope(), config_->ext_authz_failure_mode_allowed_);
       }
+      if (config_->failureModeAllowHeaderAdd()) {
+        const Http::LowerCaseString failure_mode_allow_key{absl::StrCat(
+            ThreadSafeSingleton<Http::PrefixValue>::get().prefix(), "-failure-mode-allowed")};
```

Comment:
Make this header part of `ExtAuthz::HeaderValues` to avoid repeat copy and cat at runtime.

Reply:
done, thanks.